### ISSUE TITLE
[high] Frozen audio encoder re-enters train mode — dropout/LayerDrop active throughout training

### ISSUE THREAD

**Issue by nagohachi** [nagohachi/glide#12]
## Defect

`_maybe_freeze` ([models/encoders.py#L38-L43](https://github.com/nagohachi/glide/blob/1cacf15da0d5e976c3db81648c829ee15535ab1a/src/glide/models/encoders.py#L38-L43)) sets `requires_grad=False` and calls `module.eval()` once at build time. But `Trainer.train()` calls `model.train()`, which recursively re-enables train mode on every submodule — including the "frozen" encoder. `SpeechLLM` does not override `.train()` to pin frozen submodules.

Consequence: WavLM/XLS-R LayerDrop (p≈0.05–0.1) and dropout — and Whisper dropout — stay stochastically **active** for the whole run. The frozen encoder produces noisy, dropout-scaled features at train time that differ from its eval-time behavior; with LayerDrop, entire transformer layers are randomly skipped.

## Fix

Override `SpeechLLM.train()` to keep the encoder in eval mode when it's frozen (the standard pattern), e.g.:

```python
def train(self, mode: bool = True):
    super().train(mode)
    if self._encoder_frozen:
        self.encoder.eval()
    return self
```
